File: extract_sqlite.js
const { PrismaClient } = require('@prisma/client');
const fs = require('fs');

async function extract() {
    // We need to use a separate prisma client that points to the SQLite file
    // But @prisma/client is already generated for MongoDB.
    // So we'll use a direct sqlite3 connection.
    const sqlite3 = require('sqlite3').verbose();
    const db = new sqlite3.Database('./prisma/dev.db');

    const tables = [
        'User', 'ServiceCategory', 'Service', 'Project', 'BlogPost', 
        'BlogCategory', 'Testimonial', 'TeamMember', 'FAQ', 'SiteSetting', 'SeoMeta'
    ];

    const data = {};

    for (const table of tables) {
        data[table] = await new Promise((resolve, reject) => {
            db.all(`SELECT * FROM ${table}`, (err, rows) => {
                if (err) {
                    console.log(`Table ${table} might not exist or error: ${err.message}`);
                    resolve([]);
                } else {
                    resolve(rows);
                }
            });
        });
    }

    fs.writeFileSync('extracted_data.json', JSON.stringify(data, null, 2));
    console.log('Data extracted to extracted_data.json');
    db.close();
}

extract().catch(console.error);

File: extract_sqlite.py
import sqlite3
import json
import os

def extract_data():
    db_path = os.path.join('prisma', 'dev.db')
    if not os.path.exists(db_path):
        print(f"Database not found at {db_path}")
        return

    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cursor = conn.cursor()

    # Get all tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = [row['name'] for row in cursor.fetchall() if not row['name'].startswith('_')]

    data = {}
    for table in tables:
        print(f"Extracting {table}...")
        cursor.execute(f"SELECT * FROM {table}")
        rows = cursor.fetchall()
        data[table] = [dict(row) for row in rows]

    with open('prisma/data_dump.json', 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

    print(f"Successfully extracted {len(tables)} tables to prisma/data_dump.json")
    conn.close()

if __name__ == "__main__":
    extract_data()
